### DISCUSSION TITLE
File name handling: a folder per year

### DISCUSSION THREAD

**Discussion by tido-** [jonaswinkler/paperless-ng#1538] [Support]
Hi,

Basically, we should let the database do the job for us (human). However, I would like to have at least documents per year, the rest as is from paperless-ng.
So, I tried this in `docker-compose.env`:
```
# File name handling (ein Ordner pro Jahr)  
PAPERLESS_FILENAME_FORMAT={created_year}/
``` 
Rebuilding the image. Run document_renamer. The result all files were renamed to year_##.pdf - no folder was created and the running number  0000001.pdf was gone. Like this:
2020_01.pdf
2020_02.pdf
2020_03.pdf

In the manual I saw the expression **identifier** was used but not listed as a placeholder to build the filename. Let's give it a try:
```
# File name handling (ein Ordner pro Jahr)  
PAPERLESS_FILENAME_FORMAT={created_year}/{identifier}/ 
```
Rebuilding the image. The result all files were renamed to its original state, the year completely ignored: 0000001.pdf

Any tip how I can achieve:
```
2020/
  0000001.pdf
2021/
  0000002.pdf
```
or is it simply not possible?
Thank you

**Comment by Sky-Dragon** [comment]
Hi,
you dont need the last " / " for the folder.
My File name handling is:
`PAPERLESS_FILENAME_FORMAT={correspondent}/{created_year}-{created_month}-{created_day} {document_type} {correspondent} {title} - {tag_list}`

The result are:
`
... /media/documents/archive/
1blu/
   2020-01-27 Rechnung 1blu 1blu_Beleg_XXXXXXX - Hosting.pdf
`
for more infomation here: https://paperless-ng.readthedocs.io/en/latest/advanced_usage.html#file-name-handling

I hope its help you

**Reply by tido-** [reply]
Hi, thank you.

I tried your suggestion, but I had no luck. So, I tried some more:

```
 File name handling (ein Ordner pro Jahr)
# PAPERLESS_FILENAME_FORMAT={created_year}
# PAPERLESS_FILENAME_FORMAT={created_year}/
PAPERLESS_FILENAME_FORMAT={created_year}/{none}
#PAPERLESS_FILENAME_FORMAT={created_year}/{document_type}-{identifier}
# PAPERLESS_FILENAME_FORMAT={created_year}/{identifier}/
# PAPERLESS_FILENAME_FORMAT={created_year}/{identifier}
# PAPERLESS_FILENAME_FORMAT={created_year}/{identifier}-
```
This is the only one that "worked": PAPERLESS_FILENAME_FORMAT={created_year}/{document_type}
but it was not the result I am looking for it was just a test to see if something else is wrong.

Testing got easier like this: `sudo docker-compose down && sudo docker-compose up -d && sudo docker-compose exec webserver document_renamer ` :-)

**Reply by sytone** [reply]
None of those even have the title param in them so they will be overwritten. @Sky-Dragon  pointed out the documentation for this, did you read though it? Here is the format I use, the params in the {} braces are replaced with the values for each file. if you just use {created_year} you will end up with one file per year as only that property is getting used from the file in the store. 

      # Changes the filenames paperless uses to store documents in the media directory. See File 
      # name handling for details.
      # Default is none, which disables this feature.
      # - {asn}: The archive serial number of the document, or “none”.
      # - {correspondent}: The name of the correspondent, or “none”.
      # - {document_type}: The name of the document type, or “none”.
      # - {tag_list}: A comma separated list of all tags assigned to the document.
      # - {title}: The title of the document.
      # - {created}: The full date and time the document was created.
      # - {created_year}: Year created only.
      # - {created_month}: Month created only (number 1-12).
      # - {created_day}: Day created only (number 1-31).
      # - {added}: The full date and time the document was added to paperless.
      # - {added_year}: Year added only.
      # - {added_month}: Month added only (number 1-12).
      # - {added_day}: Day added only (number 1-31).
      PAPERLESS_FILENAME_FORMAT: '{created_year}/{created_year}{created_month}{created_day}--{correspondent}--{document_type}--{tag_list}--{title}'

**Reply by tido-** [reply]
> if you just use {created_year} you will end up with one file per year

Wrong, see first message.

@sytone , if you had read my initial message and tried to understand what I try to achieve, but what the result was. Your answer would have been different, wouldn't it?
